learntools/feature_engineering/ex1.py

```diff
@@ -0,0 +1,126 @@
+from sklearn import preprocessing
+
+from learntools.core import *
+
+class TimestampFeatures(CodingProblem):
+    _vars = ['clicks', 'click_data']
+    _hint = ("With a timestamp column in a dataframe, you can get access to "
+             "datetime attibutes and functions with the `.dt` attribute. For "
+             "example `tscolumn.dt.day` will convert a timestamp column to days")
+    _solution = CS(
+    """
+    clicks = click_data.assign(day=click_times.dt.day.astype('uint8'),
+                               hour=click_times.dt.hour.astype('uint8'), 
+                               minute=click_times.dt.minute.astype('uint8'),
+                               second=click_times.dt.second.astype('uint8'))
+    """)
+    
+    def check(self, clicks, click_data):
+        click_times = click_data['click_time']
+        solution = click_data.assign(day=click_times.dt.day.astype('uint8'),
+                                     hour=click_times.dt.hour.astype('uint8'), 
+                                     minute=click_times.dt.minute.astype('uint8'),
+                                     second=click_times.dt.second.astype('uint8'))
+
+        
+
+        assert (solution.drop('attributed_time', axis=1) == 
```

Comment:
I'm getting an exception `KeyError: "['attributed_time'] not found in axis"` when trying to check my code.

I wrote my code as I would expect many users to:
```clicks = pd.DataFrame()
clicks['day'] = click_data.click_time.dt.day
clicks['hour'] = click_data.click_time.dt.hour
clicks['minute'] = click_data.click_time.dt.minute
clicks['second'] = click_data.click_time.dt.second
q_1.check()```

Having trouble seeing where `attributed_time` got created in the recommended solution.

Reply:
Ah, I see what happened. I want students to add the new columns to the original data which has the `attributed_time` feature. I'll fix this.